### PR TITLE
Remove temporary logic to allow shared paths in webhook.

### PR DESCRIPTION
**Pull request by mattmoor**
/hold

The following should land first:
 - https://github.com/knative/serving/pull/6215
 - https://github.com/knative/eventing/pull/2309

Fixes: https://github.com/knative/serving/issues/6037

### CODE REVIEW

**Comment by mattmoor** [comment]
/hold

Oops, meant to do eventing.

**Comment by mattmoor** [comment]
/hold cancel

Both landed now.

File: webhook/webhook.go [hunk 1]
```diff
@@ -114,9 +112,12 @@ func New(
 	}
 
 	// Build up a map of paths to admission controllers for routing handlers.
-	acs := map[string][]AdmissionController{}
+	acs := make(map[string]AdmissionController, len(admissionControllers))
 	for _, ac := range admissionControllers {
-		acs[ac.Path()] = append(acs[ac.Path()], ac)
+		if _, ok := acs[ac.Path()]; ok {
+			return nil, fmt.Errorf("Duplicate webhook for path: %v", ac.Path())
```
Comment (vagababov) [inline webhook/webhook.go]:
```suggestion
			return nil, fmt.Errorf("duplicate webhook for path: %v", ac.Path())
```

**Pull request merged**